### PR TITLE
Dependency refactoring

### PR DESCRIPTION
**Pull request by petersteneteg**
Added VCPKG support for all dependencies except glad, (catch2 is also not added) 
Also added a vcpkg.json file to make it easy to use with vcpkg.  And a CMakePresets.json file for common setups. The  vcpkg preset assumes that the vcpkg repo lives next to the sgct repo.

Also fixes a few errors in the examples apps

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,17 +45,19 @@ if (WIN32)
 endif ()
 
 # Exceptions for external libraries
-option(SGCT_DEP_INCLUDE_FMT "Include FMT library" ON)
```
Comment (alexanderbock) [inline CMakeLists.txt:47]: We don't align the different arguments of subsequent function calls

File: CMakeLists.txt [hunk 2]
```diff
@@ -83,11 +86,14 @@ if (UNIX)
   option(SGCT_ENABLE_STATIC_ANALYZER "Enable Static analyzer" OFF)
 endif ()
 
+#  add dummy uninstall target to avoid it beeing added by dependencies
+if(NOT TARGET uninstall)
```
Comment (alexanderbock) [inline CMakeLists.txt]: missing space after `if`

File: ext/CMakeLists.txt [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 if (SGCT_DEP_INCLUDE_GLAD) 
   add_subdirectory(glad)
-  set_property(TARGET glad PROPERTY FOLDER "Dependencies")
+  set_target_properties(glad PROPERTIES FOLDER "Dependencies")
```
Comment (alexanderbock) [inline ext/CMakeLists.txt:11]: Neato!  Didn't know about the `set_target_properties` function!

File: ext/CMakeLists.txt [hunk 4]
```diff
@@ -43,26 +46,29 @@ if (SGCT_DEP_INCLUDE_GLFW)
...
-  set_property(TARGET png PROPERTY FOLDER "Dependencies")
+  set_target_properties(png PROPERTIES FOLDER "Dependencies")
   disable_external_warnings(png)
   add_library(PNG::PNG ALIAS png)
   sgct_register_package(PNG PNG::PNG)
 endif ()
 
 if (SGCT_DEP_INCLUDE_GLM)
   add_library(glm INTERFACE)
+  add_library(glm::glm ALIAS glm )
```
Comment (alexanderbock) [inline ext/CMakeLists.txt]: Extra space before `)`

File: src/apps/stitcher/main.cpp [hunk 5]
```diff
@@ -65,7 +65,7 @@ namespace {
         float s, t;
     };
 
-    constexpr std::string_view VertexShader = R"(
+    constexpr std::string_view vertexShader = R"(
```
Comment (alexanderbock) [inline src/apps/stitcher/main.cpp]: These should be capitalized as they were before since they are `constexpr`

File: src/apps/stitcher/main.cpp [hunk 6]
```diff
@@ -79,7 +79,7 @@ void main() {
 }
 )";
 
-    constexpr std::string_view FragmentShader = R"(
+    constexpr std::string_view fragmentShader = R"(
```
Comment (alexanderbock) [inline src/apps/stitcher/main.cpp]: Same here

**Review by alexanderbock — CHANGES_REQUESTED** [review]
Getting:
```
CMake Error at apps/OpenSpace/ext/sgct/src/sgct/CMakeLists.txt:196 (find_package):
  By not providing "Findfmt.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "fmt", but
  CMake did not find one.

  Could not find a package configuration file provided by "fmt" with any of
  the following names:

    fmtConfig.cmake
    fmt-config.cmake

  Add the installation prefix of "fmt" to CMAKE_PREFIX_PATH or set "fmt_DIR"
  to a directory containing one of the above files.  If "fmt" provides a
  separate development package or SDK, be sure it has been installed.
```

when trying to Configure from OpenSpace

File: src/apps/stitcher/main.cpp (hunk 5 above)
Comment (petersteneteg) [reply src/apps/stitcher/main.cpp]: ok, needs to change the usage below then,  since that is lowercase...

File: CMakeLists.txt (hunk 1 above)
Comment (petersteneteg) [reply CMakeLists.txt:47]: fixed

File: CMakeLists.txt (hunk 2 above)
Comment (petersteneteg) [reply CMakeLists.txt]: fixed

File: ext/CMakeLists.txt (hunk 4 above)
Comment (petersteneteg) [reply ext/CMakeLists.txt]: fixed

File: src/apps/stitcher/main.cpp (hunk 5 above)
Comment (petersteneteg) [reply src/apps/stitcher/main.cpp]: modified the usage instead

File: src/apps/stitcher/main.cpp (hunk 6 above)
Comment (petersteneteg) [reply src/apps/stitcher/main.cpp]: modified the usage instead

**Comment by alexanderbock** [comment]
Still getting
```
Make Error at apps/OpenSpace/ext/sgct/src/sgct/CMakeLists.txt:184 (find_package):
  By not providing "Findfmt.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "fmt", but
  CMake did not find one.

  Could not find a package configuration file provided by "fmt" with any of
  the following names:

    fmtConfig.cmake
    fmt-config.cmake

  Add the installation prefix of "fmt" to CMAKE_PREFIX_PATH or set "fmt_DIR"
  to a directory containing one of the above files.  If "fmt" provides a
  separate development package or SDK, be sure it has been installed.
```

:(  When compiling in OpenSpace which provides its own FMT target.

Surround with
```
if (NOT TARGET fmt)
...
endif ()
```
?

**Comment by petersteneteg** [comment]
I added some handing of the case where a superproject provides targets for sgct to use. It should find fmt now

**Comment by petersteneteg** [comment]
Poke @alexanderbock I don't think your jenkins build is configured to build PRs from forks.

**Pull request closed without merging**